lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -463,9 +463,12 @@ defmodule ExUnit.Callbacks do
         {:error, :not_found}
 
       {:ok, sup} ->
-        with :ok <- Supervisor.terminate_child(sup, id),
-             :ok <- Supervisor.delete_child(sup, id),
-             do: :ok
+        with :ok <- Supervisor.terminate_child(sup, id) do
+          # If the terminated child was temporary, delete_child returns {:error, :not_found}.
+          # Since the child was successfully terminated, we treat this result as a success.
+          true = Supervisor.delete_child(sup, id) in [:ok, {:error, :not_found}]
```

Comment:
This changes the behavior of stop_supervised (without bang), as in that it case could return one of the other error status and now it will raise. What do you think? Even if we want to raise, the error message now is a bit worse.

Reply:
I did it to be consistent with the typespec, which states we can only return `:ok | {:error, :not_found}`. I also don't see any scenario where other kinds of `delete_child` errors can be returned. As [docs](https://erlang.org/doc/man/supervisor.html#delete_child-2) state:

> If successful, the function returns ok. If the child specification identified by Id exists but the corresponding child process is running or is about to be restarted, the function returns {error,running} or {error,restarting}, respectively. If the child specification identified by Id does not exist, the function returns {error,not_found}.

So this is basically just a sanity check. If this raises, I think it can only be due to some bug/wrong assumption on our behalf, and I don't see what kind of useful message could we produce in this case.